### PR TITLE
Add model file name

### PR DESCRIPTION
**Pull request by NawarA**

### CODE REVIEW

**Comment by NawarA** [comment]
@xenova here's the promised PR for #594.

I'll try to expand the solution to encoder-decoder models once I begin working with them.

**Comment by NawarA** [comment]
In the PR, you'll see my system trimmed white spaces, and the only change is:

image

**Comment by xenova** [comment]
Thanks for the PR! Just to keep the diff small, can you remove any whitespace formatting applied by your editor? There definitely are valid changes, but it will make the diff large for the PR. 🤗

**Comment by NawarA** [comment]
The PR is merged with 2.16.1 and is ready to go :)

**Comment by Th3G33k** [comment]
@xenova Those changes are already in v3. This PR should be merge into main, or close until v3 arrives

File: src/pipelines.js [hunk 1]
```diff
@@ -3107,6 +3107,7 @@ export async function pipeline(
         cache_dir = null,
         local_files_only = false,
         revision = 'main',
+        model_file_name = null
```
Comment (xenova) [inline src/pipelines.js:3110]:
```suggestion
        model_file_name = null,
```

File: src/pipelines.js [hunk 2]
```diff
@@ -3134,6 +3135,7 @@ export async function pipeline(
         cache_dir,
         local_files_only,
         revision,
+        model_file_name
```
Comment (xenova) [inline src/pipelines.js:3138]:
```suggestion
        model_file_name,
```

**Review by xenova — APPROVED** [review]
Thanks!

nits applied to keep future diffs small

**Comment by xenova** [comment]
> I'll try to expand the solution to encoder-decoder models once I begin working with them.

I'll make the necessary changes in v3 myself. For now, this is okay (and will work for all single-file models, like encoder-only and decoder-only models).

**Pull request merged**